p2p/protocol/circuitv1/relay/relay.go

```diff
@@ -64,6 +67,17 @@ func NewRelay(h host.Host, opts ...Option) (*Relay, error) {
 		}
 	}
 
+	// get a scope for memory reservations at service level
+	err := h.Network().ResourceManager().ViewService(ServiceName,
+		func(s network.ServiceScope) error {
+			var err error
+			r.scope, err = s.BeginTransaction()
+			return err
+		})
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
This block looks like it'll become boilerplate. Is there a missing API here?

Reply:
not really, most services will work at the stream level.

Reply:
on the same time, it would be very easy to define a utility method to get a txn scope for a service, and it might eliminate some boilerplate.

Where should we put it though?